### PR TITLE
Authorpage

### PR DESCRIPTION
**Pull request by FordMcLeod**
please squash me. adds author/id/posts functionality, refactored friends logic.

### CODE REVIEW

File: posts/helpers.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from .models import *
+
+def get_user( pk):
+        try:
+            return User.objects.get(pk=pk)
+        except User.DoesNotExist:
```
Comment (MichaelParadis) [inline posts/helpers.py]: This is worry some I will look into it

File: posts/urls.py [hunk 2]
```diff
@@ -9,16 +13,16 @@
     path('users/', UserView.as_view(), name='users'),
     path('author/<pk>/friends', FriendListView.as_view(), name='friendslist'),
     path('followreqs/', FollowReqListView.as_view(), name='followereqlist'),
-    # path('author/<authorid1>/friends/<service2>/author/<authorid2>', AreFriendsView.as_view(), name='arefriends'),
     path('author/<authorid1>/friends/<authorid2>', AreFriendsView.as_view(), name='arefriends'),
     path('friendrequest/', FriendRequestView.as_view(), name='friendrequest'),
-    path('follow/<authorid>', FollowView.as_view(), name='follow'),
+    path('follow/<authorid>', FollowView.as_view(),name='follow'),
```
Comment (MichaelParadis) [inline posts/urls.py:28]: end url in /

File: posts/urls.py [hunk 3]
```diff
@@ -9,16 +13,16 @@
...
     path('posts/', PostView.as_view(), name='posts'),
     path('frontend/posts/<pk>', FrontEndPostViewID.as_view(), name='frontpostid'),
     path('posts/<pk>', PostViewID.as_view(), name='postid'),
     path('posts/<post_id>/comments/', CommentViewList.as_view(), name='comments'),
     path('users/', UserView.as_view(), name='users'),
     path('approve/', AdminUserView.as_view(), name='admin-users'),
+    path('author/<authorid>/posts', FrontEndAuthorPosts.as_view(), name='frontauthorposts'),
```
Comment (MichaelParadis) [inline posts/urls.py:36]: End url in /

File: posts/helpers.py (hunk 1 above)
Comment (MichaelParadis) [reply posts/helpers.py]: Think this should be a global exception incase an invalid uuid is passed. Also we should return false and let the caller deal with the not found behavior

File: posts/views.py [hunk 4]
```diff
@@ -47,6 +52,132 @@ def put(self, request):
             return Response(serializer.data)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+class FriendRequestView(views.APIView):
```
Comment (MichaelParadis) [inline posts/views.py]: Pretty sure this is duplicating an already existing

File: posts/views.py [hunk 5]
```diff
@@ -268,6 +399,27 @@ def get(self, request):
         serializer = PostSerializer(posts, many=True)
         return Response(serializer.data)
 
+class FollowReqListView(views.APIView):
```
Comment (MichaelParadis) [inline posts/views.py]: Double added.

File: posts/views.py [hunk 6]
```diff
@@ -293,6 +445,12 @@ def get(self, request):
 
 class PostViewID(views.APIView):
 
+    def get_user(self, pk):
```
Comment (MichaelParadis) [inline posts/views.py:407]: This looks double added.

File: posts/views.py [hunk 7]
```diff
@@ -377,40 +584,3 @@ def get(self, request, post_id):
         serializer = CommentSerializer(comments, many=True)
         return Response(serializer.data)
 
-# def checkFOAF(request, userid,authorid):
```
Comment (MichaelParadis) [inline posts/views.py:412]: Did you mean to delete this?

**Review by MichaelParadis — CHANGES_REQUESTED** [review]

File: posts/views.py (hunk 7 above)
Comment (FordMcLeod) [reply posts/views.py:412]: yes

**Pull request merged**